adr/00009-assessments.md

```diff
@@ -0,0 +1,133 @@
+
+The design and user research on this part of the system is still being designed, but we understand enough to get started, and we should design the architecture in a way that is flexible enough to support future decisions.
+
+## Decision
+
+There needs to be a way of recording an assessors recommendation on an application submitted by a teacher. A recommendation could be one of decline QTS, award QTS, request a second review, or request further information (either from the teacher or from a third party).
+
+The criteria for the initial assessment of the application changes depending on the country/region rules, and certain responses given by teachers in their application (for example, if there is no name change document, there's no criteria to check one).
+
+If further information is requested, we need to support a means of requesting and receive further information for an indeterminate amount of time before a final decision is made, this means we need to allow making a recommendation (progressing the application) after the initial assessment and after further information has been received.
+
+A suggested model structure that would support this is outlined below in Rails-like pseudocode:
+
+```rb
+# Represents something that can be given a recommendation.
+# In this case it's the initial assessment, or a further information request.
+class Recommendable
+  enum recommendation: %i[
+         award
+         decline
+         review
+         further_information_from_teacher
+         further_information_from_third_party
+       ]
+end
+
+# Holds all the information about an assessment, both the initial assessments
+# and the further information requests.
+class Assessment < ApplicationRecord
+  include Recommendable
+
+  belongs_to :application_form
+
+  has_many :initial_assessments
+  has_many :further_information_requests
+
+  def finished?
+    if initial_assessments.any? { |assessment|
+         assessment.state == :not_started
+       }
+      return false
+    end
+
+    return true if recommendation == :award || recommendation == :decline
+
+    if latest_further_information_request.recommendation == :award ||
+         latest_further_information_request.recommendation == :decline
+      return true
+    end
+
+    false
+  end
+end
+
+# Represents the first stage of an assessment, which is to check the application form.
+# There will be one for each section, as the sections are dynamic depending on the
+# application form and the country/region.
+class InitialAssessment < ApplicationRecord
+  enum section: %i[
+         personal_details
+         qualifications
+         work_history
+         professional_standing
+       ]
+
+  # checked - a nullable boolean holding whether a user has changed this part of the application form
```

Comment:
I forget why we wanted this to be nullable? Doesn't defaulting to false work? Also should `a user has changed` be `a user has checked`

Reply:
Ah yeah. I remember now. Yes, I think checked is misleading. checked: false implies the section hasn't been 'checked' rather than it has but it wasn't 'passed' or 'approved'. Something like that maybe? Would using an enum be more descriptive?

Reply:
I think you're right an enum would be better, I've amended the commit with an enum. I still don't love the names, but I guess this ADR doesn't commit us to the exact naming.